src/config.py

```diff
@@ -0,0 +1,27 @@
+import os, dotenv
+from .logger import Logger
+
+class Config:
+    def __init__(self,
+                 logger=None):
+        # Load the dotenv configuration & set defaults
+        dotenv.load_dotenv()
+        self.set_defaults()
+
+        # Parse arguments & set logger
+        self.logger = logger if logger else Logger(
+            logger_name="default", logfile=self.logfile).logger
+
+    def set_defaults(self):
+        """
+        Set default global required parameters if none was found
+        """
+        # set logs file
+        self.logfile = os.getenv("OLS_LOGFILE", "logs/ols.logs")
+
+        # enable local ui?
+        self.enable_ui = True if os.getenv("OLS_ENABLE_UI", True) in [True,"True"] else False
+
+        # set default LLM model
+        self.base_completion_model = os.getenv("BASE_COMPLETION_MODEL", 
+                                               "ibm/granite-20b-instruct-v1")
```

Comment:
I struggle w/ why this particular model value is set in config/defaulted, but none of the others are?  Should we be doing this for the "yaml gen" model, "happy response model", and "classification model" here too?  Is it because this config object is not going to be shared to those other modules?

(note that those might all be the same model, but having them able to be configured separately seems logical)

Reply:
I'm maintaining current behavior with these changes, but I agree, I rather have this only come from the environment definitions and not having one hardcoded anywhere.